staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/seatsecs_test.go

```diff
@@ -17,29 +17,51 @@ limitations under the License.
 package queueset
 
 import (
-	"fmt"
 	"math"
+	"strconv"
+	"strings"
 	"testing"
 	"time"
 )
 
+// TestSeatSecondsString exercises the SeatSeconds constructor and de-constructors (String, ToFloat).
+// It uses floating-point, which can not accurately represent every real number,
+// but this test is written to only require that the floats involved are accurate +- `1/(2*ssScale)`.
+// This test does not require editing when the value of `ssScale` changes.
+// The check on fraction digits is the only thing that requires `ssScale` to be a power of ten.
 func TestSeatSecondsString(t *testing.T) {
-	digits := math.Log10(ssScale)
-	expectFmt := fmt.Sprintf("%%%d.%dfss", int(digits+2), int(digits))
+	digits := int(math.Round(math.Log10(ssScale)))
 	testCases := []struct {
 		ss     SeatSeconds
-		expect string
+		expect float64
 	}{
-		{ss: SeatSeconds(1), expect: fmt.Sprintf(expectFmt, 1.0/ssScale)},
-		{ss: 0, expect: "0.00000000ss"},
-		{ss: SeatsTimesDuration(1, time.Second), expect: "1.00000000ss"},
-		{ss: SeatsTimesDuration(123, 100*time.Millisecond), expect: "12.30000000ss"},
-		{ss: SeatsTimesDuration(1203, 10*time.Millisecond), expect: "12.03000000ss"},
+		{ss: SeatSeconds(1), expect: 1.0 / ssScale},
+		{ss: SeatSeconds(ssScale - 1), expect: (ssScale - 1) / ssScale},
+		{ss: 0, expect: 0},
+		{ss: SeatsTimesDuration(1, time.Second), expect: 1},
+		{ss: SeatsTimesDuration(123, 100*time.Millisecond), expect: 12.3},
+		{ss: SeatsTimesDuration(1203, 10*time.Millisecond), expect: 12.03},
 	}
 	for _, testCase := range testCases {
 		actualStr := testCase.ss.String()
-		if actualStr != testCase.expect {
-			t.Errorf("SeatSeonds(%d) formatted as %q rather than expected %q", uint64(testCase.ss), actualStr, testCase.expect)
+		actualStrLen := len(actualStr)
+		actualSuffix := actualStr[actualStrLen-2:]
+		if actualSuffix != "ss" {
```

Comment:
I'm against these types of types.
The tests are supposed to be as simple as possible to avoid bugs in tests themselfes.

Given that this in particular is testing if the returned string looks as it should, what we should do is:
- have an `expectString` field in the testCase
- simply compare that with with what ss.String() returns